### PR TITLE
add cpumanager

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

#### What type of PR is this?
Add cpumanager.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: pkg/ensurance/cm/cpumanager/cpu_manager.go [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+	cm.policy = NewStaticPolicy(topo, getPodFunc)
+
+	cm.state, err = cpumanagerstate.NewCheckpointState(
+		stateFileDirectory,
+		cpuManagerStateFileName,
+		cm.policy.Name(),
+		initialContainers,
+	)
+	if err != nil {
+		klog.ErrorS(err, "Could not initialize checkpoint state, please drain node and remove policy state file")
```
Comment (mfanjie) [inline pkg/ensurance/cm/cpumanager/cpu_manager.go:195]: why drain node is required on error? the state file is inside the agent pod not on the host right?

File: cmd/crane-agent/app/options/option.go [hunk 2]
```diff
@@ -53,11 +61,14 @@ func (o *Options) AddFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&o.RuntimeEndpoint, "runtime-endpoint", "", "The runtime endpoint docker: unix:///var/run/dockershim.sock, containerd: unix:///run/containerd/containerd.sock, cri-o: unix:///run/crio/crio.sock, k3s: unix:///run/k3s/containerd/containerd.sock.")
 	flags.StringVar(&o.CgroupDriver, "cgroup-driver", "cgroupfs", "Driver that the kubelet uses to manipulate cgroups on the host.  Possible values: 'cgroupfs', 'systemd'. Default to 'cgroupfs'")
 	flags.StringVar(&o.SysPath, "sys-path", "/sys", "Path to /sys dir.")
+	flags.StringVar(&o.KubeletRootPath, "kubelet-root-path", o.KubeletRootPath, "Path to the kubelet root directory.")
```
[the remark this answers is not in the document]
Comment (Garrybest) [reply cmd/crane-agent/app/options/option.go:64]: Usually this path is `/var/lib/kubelet` in host, if we use default value in container, we should mount `/var` from host to container, is it appropriate?

File: pkg/agent/agent.go [hunk 3]
```diff
@@ -84,15 +87,29 @@ func NewAgent(ctx context.Context,
...
-	if craneCpuSetManager := utilfeature.DefaultFeatureGate.Enabled(features.CraneCpuSetManager); craneCpuSetManager {
-		cpuManager := cm.NewAdvancedCpuManager(podInformer, runtimeEndpoint, cadvisorManager)
-		exclusiveCPUSet = cpuManager.GetExclusiveCpu
-		managers = appendManagerIfNotNil(managers, cpuManager)
+	exclusiveCPUSet := cpumanager.DefaultExclusiveCPUSet
+	if utilfeature.DefaultFeatureGate.Enabled(features.CraneNodeResourceTopology) {
+		if err := agent.CreateNodeResourceTopology(sysPath); err != nil {
+			return nil, err
+		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.CraneCPUManager) {
```
[the remark this answers is not in the document]
Comment (Garrybest) [reply pkg/agent/agent.go:103]: Yes, in that case only nrt is collected, cpumanager does nothing to allocate cpuset.

File: pkg/ensurance/cm/cpumanager/cpu_manager.go (hunk 1 above)
Comment (Garrybest) [reply pkg/ensurance/cm/cpumanager/cpu_manager.go:195]: We mount kubelet root path into container, so the crane state file will be also created in host path. It seems like we don't need to drain node, I just copy the code from `k/k`. Let me make a change here.

**Review by mfanjie — APPROVED** [review]
LGTM

**Pull request merged**